src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -2621,6 +2622,47 @@ class misc_commandscript : public CommandScript
         handler->GetSession()->SendShowMailBox(player->GetGUID());
         return true;
     }
+    
+    static bool HandleNearGraveyard(ChatHandler* handler, char const* /*args*/)
+    {
+        Player* player = handler->GetSession()->GetPlayer();
+
+        float x = player->GetPositionX();
+        float y = player->GetPositionY();
+        float z = player->GetPositionZ();
+
+        const WorldSafeLocsEntry* nearestLoc = nullptr;
+        bool isNear = false;
+        float distNearest = 20000.0f;
+        
+        for (uint32 i = 0; i < sWorldSafeLocsStore.GetNumRows(); ++i)
+        {
+            const WorldSafeLocsEntry* loc = sWorldSafeLocsStore.LookupEntry(i);
+            if (loc && loc->map_id == player->GetMapId())
+            {
+                float dist = (loc->x - x) * (loc->x - x) + (loc->y - y) * (loc->y - y) + (loc->z - z) * (loc->z - z);
+                if (isNear)
+                {
+                    if (dist < distNearest)
+                    {
+                        distNearest = dist;
+                        nearestLoc = loc;
+                    }
+                }
+                else
+                {
+                    isNear = true;
+                    distNearest = dist;
+                    nearestLoc = loc;
+                }
+            }
+        }
+
+        if (nearestLoc)
+            handler->PSendSysMessage(LANG_NEARGRAVEYARD, nearestLoc->ID, nearestLoc->x, nearestLoc->y, nearestLoc->z);
```

Comment:
is there a case when a graveyard could not be found so that we need to tell the user with another message ? or that shouldn't matter (arenas/instances maybe ?)